\documentclass[a4,center,fleqn]{NAR}

% Enter dates of publication
\copyrightyear{2017}
\pubdate{XX September XXXX}
\pubyear{XXXX}
\jvolume{XX}
\jissue{XX}

%\articlesubtype{This is the article type (optional)}
% \usepackage{xr-hyper}
% \externaldocument[ ]{riboSeed_NAR_suppl}

\usepackage{caption}
\graphicspath{ {./riboFigs/} } % set the path to the figures

\begin{document}

\title{riboSeed: leveraging prokaryotic genomic architecture to assemble across ribosomal regions}

\author{%
  Nicholas R. Waters\,$^{1,2}$
  Florence Abram\,$^{1}$
  Fiona Brennan\,$^{1,3}$
  Ashleigh Holmes\,$^{4}$
  and Leighton Pritchard$^{2}$
  \footnote{To whom correspondence should be addressed.
      Tel: +44 (0) 1382 568827; E-mail: leighton.pritchard@hutton.ac.uk
  }
}
\address{
  $^{1}$Department of Microbiology, School of Natural Sciences, National University of Ireland, Galway, Ireland \\
  $^{2}$Information and Computational Sciences, James Hutton Institute, Invergowrie, Dundee DD2 5DA, Scotland\\
  $^{3}$Soil and Environmental Microbiology, Environmental Research Centre, Teagasc, Johnstown Castle, Wexford, Ireland\\
  $^{4}$Cell and Molecular Sciences, James Hutton Institute, Invergowrie, Dundee DD2 5DA, Scotland\\
}
% $^{1}$Affiliation of Corresponding Author
% and
% $^{2}$Affiliation of Both Co-Authors}
% Affiliation must include:
% Department name, institution name, full road and district address,
% state, Zip or postal code, country

\history{%
Received January 1, 2009;
Revised February 1, 2009;
Accepted March 1, 2009}

\maketitle

\begin{abstract}
  \input{main_tex/abstract}
\end{abstract}

\section{Introduction}

\input{main_tex/figncbi}

\input{main_tex/intro1}

% **************************************************************
% Keep this command to avoid text of first page running into the
% first page footnotes
\enlargethispage{-65.1pt}
% **************************************************************

\input{main_tex/tabfinishers}

\input{main_tex/intro2}

\input{main_tex/figsummary}

\section{MATERIALS AND METHODS}
\input{main_tex/usage1}

\input{main_tex/figcolientropy}

\input{main_tex/usage2}

\input{main_tex/avail}

\section{RESULTS}

\subsection*{Characteristics of rDNA flanking regions}

\input{main_tex/resentropy}

\subsection*{Simulated reads with artificial chromosome}

\input{main_tex/figmauve}

\input{main_tex/resartificial}




\subsection*{Effect of reference sequence identity on riboSeed performance}

\input{main_tex/ressimread1}

\input{main_tex/figdegen}

\input{main_tex/ressimread2}

\input{main_tex/figcolikleb}

\subsection*{Simulated reads with \textit{E. coli} and \textit{K. pneumoniae} genomes}

\input{main_tex/ressimread3}

\subsection*{Benchmarking against hybrid sequencing and assembly}

\input{main_tex/reshybrid}

\input{main_tex/tabsnps}

\subsection*{Case Study: Closing the assembly of \textit{S. aureus} UAMS-1}

\input{main_tex/resuams1}

\subsection*{Benchmarking against GAGE-B Datasets}

\input{main_tex/tabbenchmark}

\input{main_tex/resbenchmark}

\section*{DISCUSSION}

\input{main_tex/discussion}

\section*{CONCLUSION}

\input{main_tex/conclusion}

\section{FUNDING}
The work was supported by a joint studentship between The James Hutton Institute, Dundee, Scotland, and the National University of Ireland, Galway, Ireland.


\section{ACKNOWLEDGEMENTS}
We thank Anton Korobeynikov for his recommendations on optimizing SPAdes. Yoann Augagneur, Shaun Brinsmade, and Mohamed Sassi graciously provided access to the \textit{S. aureus} UAMS-1 genome sequencing data.  Additional computational resources were provided by CLIMB \cite{Connor2016}. We thank the Bioconda (\url{https://bioconda.github.io/}) community for their support.


\subsection{Conflict of interest statement.}
None declared.

\bibliographystyle{unsrt}
\bibliography{riboSeed_refs}

\end{document}
