

%  I need to do this to deal with the page breaks
\begin{figure}[]
  \centering
  \foreach \x in {1,2,...,5}{
    \begin{subfigure}[b]{.45\textwidth}
      \includegraphics[width=0.95\textwidth]{entropy_results_figs/\x_genome.png}
    \caption{}
    \end{subfigure}
    \begin{subfigure}[b]{.45\textwidth}
      \includegraphics[width=0.95\textwidth]{entropy_results_figs/\x_gene.png}
    \caption{}
    \end{subfigure}
  }
  \label{fig:ent_gage}
\end{figure}

% And each set here after
\begin{figure}[]\ContinuedFloat
  \centering
  \foreach \x in {6,7,...,10}
  {
    \begin{subfigure}[b]{.45\textwidth}
      \includegraphics[width=0.95\textwidth]{entropy_results_figs/\x_genome.png}
    \caption{}
    \end{subfigure}
    \begin{subfigure}[b]{.45\textwidth}
      \includegraphics[width=0.95\textwidth]{entropy_results_figs/\x_gene.png}
    \caption{}
    \end{subfigure}
  }
\end{figure}

% 11-15
% 12 Sakai was omitted, as it was present in manuscript
\begin{figure}[]\ContinuedFloat
  \centering
  \foreach \x in {11,12,13,15}
  {
    \begin{subfigure}[b]{.45\textwidth}
      \includegraphics[width=0.95\textwidth]{entropy_results_figs/\x_genome.png}
    \caption{}
    \end{subfigure}
    \begin{subfigure}[b]{.45\textwidth}
      \includegraphics[width=0.95\textwidth]{entropy_results_figs/\x_gene.png}
    \caption{}
    \end{subfigure}
  }
\end{figure}

% And each set here after
% 16 was omitted (duplicate P aeruginosa)
\begin{figure}[]\ContinuedFloat
  \centering
  \foreach \x in {17,...,21}
  {
    \begin{subfigure}[b]{.45\textwidth}
      \includegraphics[width=0.95\textwidth]{entropy_results_figs/\x_genome.png}
    \caption{}
    \end{subfigure}
    \begin{subfigure}[b]{.45\textwidth}
      \includegraphics[width=0.95\textwidth]{entropy_results_figs/\x_gene.png}
    \caption{}
    \end{subfigure}
  }
\end{figure}

%  I need to do this to deal with the page breaks
\begin{figure}[]\ContinuedFloat
  \centering
  \foreach \x in {21,22}{
    \begin{subfigure}[b]{.45\textwidth}
      \includegraphics[width=0.95\textwidth]{entropy_results_figs/\x_genome.png}
    \caption{}
    \end{subfigure}
    \begin{subfigure}[b]{.45\textwidth}
      \includegraphics[width=0.95\textwidth]{entropy_results_figs/\x_gene.png}
    \caption{}
    \end{subfigure}
  }
  \label{fig:ent_gage}
  \caption{riboScan.py, riboSelect.py, and riboSnag.py were run on all the genomes used as references for \textit{de fere novo} assemblies. Consensus alignment depth (grey bars) and Shannon entropy (black points, smoothed entropy as red line) for aligned rDNA regions.  Similar to Figure 3 in the main text, for each genome, a gene neighboring the first rDNA operon was identified, and used to extract homologous rDNA operons from up to 25 other isolates at the species level. In most cases, the entropy is lower in homologous rDNAs than across all the rDNAs in a given genome.  For strains with a low number of complete genomes for comparison available, entropy may be artificially increased (see \textit{Mycoplasma hominis}) or decreased (\textit{Helicobacter cinaedi}). A baseline entropy of greater than 0 may indicate equal distribution of two alleles of the operon either within a genome or across genomes.}
\end{figure}
