To create a small dataset for testing, we extracted all 7 distinct rDNA regions from the \textit{E. coli} Sakai genome (BA000007.2), including 5kb upstream and downstream flanking sequence, using the tools \texttt{scan}, \texttt{select} and \texttt{snag}. Those regions were concatenated to produce a \textasciitilde100kb artificial test chromosome (see supplementary methods). pIRS \cite{Hu2012} was used to generate simulated reads (100bp, 300bp inserts, stdev 10, 30-fold coverage, built-in error profile) from this test chromosome. These reads were assembled using riboSeed, using the \textit{E. coli} MG1655 genome (NC\_000913.3) as a reference.  Simulation was repeated 8 times to assess variability of method performance on alternative read sets generated from the same source sequence; Figure \ref{fig:artificial} shows a Mauve visualization of a representative run.


\textit{de fere novo} assembly bridged 4 of the 7 rDNA regions in the artificial chromosome, while \textit{de novo} assembly failed to bridge any (Figure S3). To illustrate how choice of reference sequence determines correct assembly through rDNA, we ran riboSeed with the same \textit{E. coli} reads using pseudocontigs derived from the \textit{Klebsiella pneumoniae} HS11286 (CP003200.1) reference genome \cite{Liu2012}. \textit{de fere novo} assembly with pseudocontigs from \textit{K. pneumoniae} failed, as the reference is too divergent from the reads.
