To obtain an estimate of substitution rate for the \textit{E. coli} strains used above, Parsnp \cite{Treangen2014} and Gingr \cite{Treangen2014} were used to identify SNPs in the 25 genomes used in Figure \ref{fig:entropy}, with respect to the same region in \textit{E. coli} Sakai. An average substitution rate of $\approx$ 3.5 substitutions per kb was observed. Compared to the results from simulated genomes, we expect successful riboSeed performance under the model of mutated flanking regions, and partial success under the model of substitutions throughout the region.


Figure \ref{fig:degen} indicates that the greater the similarity of the reference sequence to the genome being assembled, the greater the likelihood of correctly assembling all rDNA regions. When mutating only flanking regions (Figure \ref{fig:degen}), which more closely resembles the relative sustitution frequencies of the rDNA regions, the procedure correctly assembles rDNAs with tolerance to substitution frequencies up to approximately 30 substitutions per kb.  With the widely-adopted average nucleotide identity species boundary of 95\% \cite{Goris2007a}, we anticipate that riboSeed should correctly place and assemble most rDNA regions when using a complete reference genome of the same species, and that reasonable success will be achieved even when using a more distantly-related reference.
