\subsection*{3. Assessment and Visualization}

\textbf{\texttt{score}}\\
\texttt{score} extracts the regions flanking rDNAs in the reference and in assemblies generated by riboSeed. Flanking regions from an assembly are matched with reference flanking regions using BLASTn. Depending on the ordering of the matches, assembled junctions are called as correct, incorrect, or ambiguous based on the criteria outlined below.\\
\\
\textbf{\texttt{snag}}\\
\texttt{snag} is a helper tool to produce diagnostics and visualisation of rDNA sequences in the reference genome. Using the clustering generated by \texttt{select}, sequences for the clusters are extracted from the genome, aligned, and Shannon entropy \cite{Schmitt1997} plotted with consensus depth for each position in the alignment.\\
\\
\textbf{\texttt{swap}}\\
We recommend assessing the performance of the riboSeed pipeline visually using Mauve \cite{Darling2004,Darling2011}, Gingr \cite{Treangen2014}, or similar genome assembly visualizer to compare reference, \textit{de novo}, and \textit{de fere novo} assemblies. If contigs appear to be incorrectly joined, the offending \textit{de fere novo} contig can be replaced with syntenic contigs from the \textit{de novo} assembly using the \texttt{swap} script.\\
\\
\textbf{\texttt{stack}}\\
\texttt{stack} uses bedtools \cite{Quinlan2010} and samtools \cite{Li2009} to compare depth of coverage of reads aligning to the reference genome in the rDNA regions to randomly sampled regions elsewhere in the reference genome. \texttt{stack} takes output from \texttt{scan}, and a BAM file of reads that map to the reference. If the number of \texttt{scan}-annotated rDNAs matches the number of rDNAs in the sequenced isolate, the coverage depths within the rDNAs will be similar to the coverage in other locations in the genome. If the coverage of rDNA regions sufficiently exceeds the average coverage elsewhere in the genome, this may indicate that the reference strain has fewer rDNAs than the sequenced isolate. In this case, using an alternative reference genome may produce improved results.
